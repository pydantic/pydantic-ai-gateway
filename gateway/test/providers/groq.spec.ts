import Groq from 'groq-sdk'
import { describe, expect } from 'vitest'
import { test } from '../setup'

describe('groq', () => {
  test('should call groq via gateway', async ({ gateway }) => {
    const { fetch, otelBatch } = gateway
    const client = new Groq({ apiKey: 'healthy', baseURL: 'https://example.com/groq', fetch })

    const completion = await client.chat.completions.create({
      model: 'llama-3.3-70b-versatile',
      messages: [
        { role: 'developer', content: 'You are a helpful assistant.' },
        { role: 'user', content: 'What is the capital of France?' },
      ],
      top_p: 0.95,
      temperature: 0.5,
      stop: ['potato'],
      max_completion_tokens: 1024,
    })
    expect(completion).toMatchSnapshot('llm')
    expect(otelBatch, 'otelBatch length not 1').toHaveLength(1)
    expect(JSON.parse(otelBatch[0]!).resourceSpans?.[0].scopeSpans?.[0].spans?.[0]?.attributes).toMatchSnapshot('span')
  })
})
