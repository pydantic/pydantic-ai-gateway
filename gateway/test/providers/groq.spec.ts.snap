// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`groq > should call groq via gateway > llm 1`] = `
{
  "choices": [
    {
      "finish_reason": "stop",
      "index": 0,
      "logprobs": null,
      "message": {
        "content": "The capital of France is Paris.",
        "role": "assistant",
      },
    },
  ],
  "created": 1758119096,
  "id": "chatcmpl-bf4d0706-9871-4926-bcac-0158874195d2",
  "model": "llama-3.3-70b-versatile",
  "object": "chat.completion",
  "service_tier": "on_demand",
  "system_fingerprint": "fp_3f3b593e33",
  "usage": {
    "completion_time": 0.002177697,
    "completion_tokens": 8,
    "prompt_time": 0.011647025,
    "prompt_tokens": 42,
    "pydantic_ai_gateway": {
      "cost_estimate": 0.0000311,
    },
    "queue_time": 0.086600393,
    "total_time": 0.013824722,
    "total_tokens": 50,
  },
  "usage_breakdown": null,
  "x_groq": {
    "id": "req_01k5c10vm6f3ebj27ah02hjqy3",
  },
}
`;

exports[`groq > should call groq via gateway > span 1`] = `
[
  {
    "key": "logfire.msg",
    "value": {
      "stringValue": "chat llama-3.3-70b-versatile",
    },
  },
  {
    "key": "logfire.json_schema",
    "value": {
      "stringValue": "{"type":"object","properties":{"gen_ai.operation.name":{"type":"string"},"gen_ai.request.model":{"type":"string"},"gen_ai.system":{"type":"string"},"gen_ai.request.max_tokens":{"type":"number"},"gen_ai.request.top_k":{},"gen_ai.request.top_p":{"type":"number"},"gen_ai.request.temperature":{"type":"number"},"gen_ai.request.stop_sequences":{},"gen_ai.request.seed":{},"gen_ai.response.finish_reasons":{},"gen_ai.response.id":{"type":"string"},"gen_ai.input.messages":{},"gen_ai.output.messages":{},"gen_ai.system_instructions":{},"http.response.status_code":{"type":"number"},"http.request.body.text":{"type":"string"},"http.response.body.text":{"type":"string"},"gen_ai.response.model":{"type":"string"},"gen_ai.usage.input_tokens":{"type":"number"},"gen_ai.usage.cache_read_tokens":{},"gen_ai.usage.cache_write_tokens":{},"gen_ai.usage.output_tokens":{"type":"number"},"gen_ai.usage.input_audio_tokens":{},"gen_ai.usage.cache_audio_read_tokens":{},"gen_ai.usage.output_audio_tokens":{},"events":{}}}",
    },
  },
  {
    "key": "logfire.level_num",
    "value": {
      "intValue": 9,
    },
  },
  {
    "key": "gen_ai.operation.name",
    "value": {
      "stringValue": "chat",
    },
  },
  {
    "key": "gen_ai.request.model",
    "value": {
      "stringValue": "llama-3.3-70b-versatile",
    },
  },
  {
    "key": "gen_ai.system",
    "value": {
      "stringValue": "groq",
    },
  },
  {
    "key": "gen_ai.request.max_tokens",
    "value": {
      "intValue": 1024,
    },
  },
  {
    "key": "gen_ai.request.top_k",
    "value": {},
  },
  {
    "key": "gen_ai.request.top_p",
    "value": {
      "doubleValue": 0.95,
    },
  },
  {
    "key": "gen_ai.request.temperature",
    "value": {
      "doubleValue": 0.5,
    },
  },
  {
    "key": "gen_ai.request.stop_sequences",
    "value": {
      "arrayValue": {
        "values": [
          {
            "stringValue": "potato",
          },
        ],
      },
    },
  },
  {
    "key": "gen_ai.request.seed",
    "value": {},
  },
  {
    "key": "gen_ai.response.finish_reasons",
    "value": {
      "arrayValue": {
        "values": [
          {
            "stringValue": "stop",
          },
        ],
      },
    },
  },
  {
    "key": "gen_ai.response.id",
    "value": {
      "stringValue": "chatcmpl-bf4d0706-9871-4926-bcac-0158874195d2",
    },
  },
  {
    "key": "gen_ai.input.messages",
    "value": {
      "arrayValue": {
        "values": [
          {
            "kvlistValue": {
              "values": [
                {
                  "key": "role",
                  "value": {
                    "stringValue": "system",
                  },
                },
                {
                  "key": "parts",
                  "value": {
                    "arrayValue": {
                      "values": [
                        {
                          "kvlistValue": {
                            "values": [
                              {
                                "key": "type",
                                "value": {
                                  "stringValue": "text",
                                },
                              },
                              {
                                "key": "content",
                                "value": {
                                  "stringValue": "You are a helpful assistant.",
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                },
              ],
            },
          },
          {
            "kvlistValue": {
              "values": [
                {
                  "key": "role",
                  "value": {
                    "stringValue": "user",
                  },
                },
                {
                  "key": "parts",
                  "value": {
                    "arrayValue": {
                      "values": [
                        {
                          "kvlistValue": {
                            "values": [
                              {
                                "key": "type",
                                "value": {
                                  "stringValue": "text",
                                },
                              },
                              {
                                "key": "content",
                                "value": {
                                  "stringValue": "What is the capital of France?",
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                },
              ],
            },
          },
        ],
      },
    },
  },
  {
    "key": "gen_ai.output.messages",
    "value": {
      "arrayValue": {
        "values": [
          {
            "kvlistValue": {
              "values": [
                {
                  "key": "role",
                  "value": {
                    "stringValue": "assistant",
                  },
                },
                {
                  "key": "parts",
                  "value": {
                    "arrayValue": {
                      "values": [
                        {
                          "kvlistValue": {
                            "values": [
                              {
                                "key": "type",
                                "value": {
                                  "stringValue": "text",
                                },
                              },
                              {
                                "key": "content",
                                "value": {
                                  "stringValue": "The capital of France is Paris.",
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                },
                {
                  "key": "finish_reason",
                  "value": {
                    "stringValue": "stop",
                  },
                },
              ],
            },
          },
        ],
      },
    },
  },
  {
    "key": "gen_ai.system_instructions",
    "value": {},
  },
  {
    "key": "http.response.status_code",
    "value": {
      "intValue": 200,
    },
  },
  {
    "key": "http.request.body.text",
    "value": {
      "stringValue": "{
  "model": "llama-3.3-70b-versatile",
  "messages": [
    {
      "role": "developer",
      "content": "You are a helpful assistant."
    },
    {
      "role": "user",
      "content": "What is the capital of France?"
    }
  ],
  "top_p": 0.95,
  "temperature": 0.5,
  "stop": [
    "potato"
  ],
  "max_completion_tokens": 1024
}",
    },
  },
  {
    "key": "http.response.body.text",
    "value": {
      "stringValue": "{"id":"chatcmpl-bf4d0706-9871-4926-bcac-0158874195d2","object":"chat.completion","created":1758119096,"model":"llama-3.3-70b-versatile","choices":[{"index":0,"message":{"role":"assistant","content":"The capital of France is Paris."},"logprobs":null,"finish_reason":"stop"}],"usage":{"queue_time":0.086600393,"prompt_tokens":42,"prompt_time":0.011647025,"completion_tokens":8,"completion_time":0.002177697,"total_tokens":50,"total_time":0.013824722,"pydantic_ai_gateway":{"cost_estimate":0.0000311}},"usage_breakdown":null,"system_fingerprint":"fp_3f3b593e33","x_groq":{"id":"req_01k5c10vm6f3ebj27ah02hjqy3"},"service_tier":"on_demand"}",
    },
  },
  {
    "key": "gen_ai.response.model",
    "value": {
      "stringValue": "llama-3.3-70b-versatile",
    },
  },
  {
    "key": "gen_ai.usage.input_tokens",
    "value": {
      "intValue": 42,
    },
  },
  {
    "key": "gen_ai.usage.cache_read_tokens",
    "value": {},
  },
  {
    "key": "gen_ai.usage.cache_write_tokens",
    "value": {},
  },
  {
    "key": "gen_ai.usage.output_tokens",
    "value": {
      "intValue": 8,
    },
  },
  {
    "key": "gen_ai.usage.input_audio_tokens",
    "value": {},
  },
  {
    "key": "gen_ai.usage.cache_audio_read_tokens",
    "value": {},
  },
  {
    "key": "gen_ai.usage.output_audio_tokens",
    "value": {},
  },
  {
    "key": "events",
    "value": {},
  },
]
`;
